react-accordion-component/src/accordion.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+class Accordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {htmlDescOpen: false, cssDescOpen: false, jsDescOpen: false};
+    this.htmlClick = this.htmlClick.bind(this);
+    this.cssClick = this.cssClick.bind(this);
+    this.jsClick = this.jsClick.bind(this);
+  }
+  htmlClick() {
+    if (this.state.htmlDescOpen === true) {
+      this.setState({ htmlDescOpen: false, cssDescOpen: false, jsDescOpen: false });
+    } else {
+      this.setState({ htmlDescOpen: true, cssDescOpen: false, jsDescOpen: false });
+    }
+  }
+  cssClick() {
+    if (this.state.cssDescOpen === true) {
+      this.setState({ htmlDescOpen: false, cssDescOpen: false, jsDescOpen: false });
+    } else {
+      this.setState({ htmlDescOpen: false, cssDescOpen: true, jsDescOpen: false});
+    }
+  }
+  jsClick() {
+    if (this.state.jsDescOpen === true) {
+      this.setState({ htmlDescOpen: false, cssDescOpen: false, jsDescOpen: false });
+    } else {
+      this.setState({ htmlDescOpen: false, cssDescOpen: false, jsDescOpen: true });
+    }
+  }
```

Comment:
Check the constraint section of the challenge. The component shouldn't know that it's receiving html, css, and javascript at all. 

You'll need to use `.map()` for this one. Try keeping track of the index of the open topic

```js
this.state = {
  openTopicIndex: null
};
```

Open the matching topic in the array. `.map()` callback takes the index too.

```js
this.props.topics.map((topic, index) => {
  // ??? you'll need conditional logic in here
});
```

if the clicked topic matches the open topic, then set the state back to `null` to close them all.